src/algorithms/synthesis/encoding.cpp

```diff
@@ -126,21 +101,57 @@ namespace syrec {
             output = encoding[output];
         }
 
-        return {encoding, additionalLines};
+        return encoding;
     }
 
-    auto augmentWithConstants(TruthTable& tt, std::size_t const& nBits, bool dc) -> void {
-        const auto ancillaBits = nBits - tt.nOutputs();
+    auto augmentWithConstants(TruthTable& tt, std::size_t const& nBits, bool appendZero) -> void {
+        const auto requiredOutConstants = nBits - tt.nOutputs();
+        const auto requiredInConstants  = nBits - tt.nInputs();
 
         for (auto& [input, output]: tt) {
-            // add necessary constant inputs to the outputs based on the total number of bits (nBits).
-            if (!dc) {
-                for (auto i = 0U; i < ancillaBits; i++) {
-                    output.insertZero();
+            const auto oldGarbageVec         = tt.getGarbage();
+            const auto currentGarbageVecSize = oldGarbageVec.size();
+
+            if (appendZero) {
+                // based on the requiredOutConstants, zeros are appended to the outputs.
+                for (auto i = 0U; i < requiredOutConstants; i++) {
+                    // based on the requiredOutConstants, zeros are appended to the outputs.
+                    output.emplace_back(false);
+
+                    if (currentGarbageVecSize != nBits) {
+                        // reset and resize the garbage.
+                        tt.resetGarbage();
+                        tt.resizeGarbage(currentGarbageVecSize + 1);
+
+                        for (auto cg = 0U; cg < currentGarbageVecSize; cg++) {
+                            if (oldGarbageVec[cg]) {
+                                // storing the old garbage bits in proper order.
+                                tt.setGarbage(cg + 1);
+                            }
+                        }
+                        // setting the newly added garbage bits.
+                        tt.setGarbage(0);
+                    }
```

Comment:
This is an incredibly wasteful and verbose way of just appending a single Boolean to the existing vectors of garbage values. If you provide a `[[nodiscard]] auto& getGarbage() -> std::vector<bool>&` method and similar for the constants, you don't need to write extra wrapper for that, but can just use `tt.getGarbage().insert(0, false)`